Better explanation of using custom/self-signed certificates

<!--
    If you are an Enterprise customer, you might want to head over to our
    [Customer Support Portal](https://support.neo4j.com/).
-->

<!--
    If you are unsure about the usage of the driver, this issue tracker is not
    the right place to ask. Please use one of the following channels instead:

    - Neo4j Online Community: https://community.neo4j.com/
    - StackOverflow: https://stackoverflow.com/questions/tagged/neo4j+python
    - Discord: https://discord.gg/neo4j
    - Neo4j Support Portal if you are an enterprise customer:
      https://support.neo4j.com/

    or check the documentation:

    - API documentation: https://neo4j.com/docs/api/python-driver/current/
    - Driver manual: https://neo4j.com/docs/python-manual/current/
-->

<!--
    Make sure you've searched the issue tracker for similar issues first.
-->

# Feature Request
<!-- a brief description of what's requested -->

I'm trying to understand how to properly use Custom Certificates (or Self Signed Certificates).
The documentation and related error messages are confusing.

The documentation here says that if I want to use a custom/self-signed certificate I need to use the `neo4j+ssc` scheme.
https://neo4j.com/docs/api/python-driver/5.28/api.html#uri

However when I do that I get an error.

`neo4j.exceptions.ConfigurationError: The config settings "encrypted", "trust", "trusted_certificates", and "ssl_context" can only be used with the URI schemes ['bolt', 'neo4j']. Use the other URI schemes ['bolt+ssc', 'bolt+s', 'neo4j+ssc', 'neo4j+s'] for setting encryption settings.`

I've tried using both SSL `ssl_context` and `trusted_certificates`, they both throw the same error.

MRE(ish) below 👇 

```python
import pathlib
import ssl

import neo4j
from neo4j import GraphDatabase

from my_app import CUSTOM_CERT_PATH # pathlib.Path object

def get_config() -> dict:
   return {"user_name": "username", "password": "pw", "uri": "neo4j+ssc://12345.databases.neo4j.io"}

def get_neo4j_driver_ssl_context(config: dict[str, str]) -> neo4j.Driver:
    return GraphDatabase.driver(
        config["uri"],
        auth=(config["username"], config["password"]),
        ssl_context=ssl.create_default_context(cafile=CUSTOM_CERT_PATH),
    )

def get_neo4j_driver_custom_ca(config: dict[str, str]) -> neo4j.Driver:
    return GraphDatabase.driver(
        config["uri"],
        auth=(config["username"], config["password"]),
        trusted_certificates=neo4j.TrustCustomCAs(str(CUSTOM_CERT_PATH))
    )

if __name__ == "__main__":
  config = get_config()
  ssl_ctx_driver = get_neo4j_driver_ssl_context(config)
  ssl_ctx_driver.verify_connectivity()

  custom_ca_driver = get_neo4j_driver_custom_ca(config)
  custom_ca_driver.verify_connectivity()
```

I should note I've used this exact cert file with multiple other (non Aura/Neo4J) services without issue.

The DB in question is on/in Aura if the uri didn't make that clear.

I can of course connect when I drop the the `+ssc` scheme (example: `neo4j+ssc://12345.databases.neo4j.io`) but it goes without saying I don't want to to use the driver without encryption.

### Version Info
```
neo4j v5.28.2
Python 3.13.5
```

# Pitch
<!-- elaborate on the feature request and why it would be beneficial -->

Please update or expand your documentation around using custom certificates along with examples, or fix the underlying issue that prevents using self signed certs in the manor that is described.

Here's the more detailed stacktrace.

Notice 👉  `parsed = ParseResult(scheme='neo4j+ssc', netloc='12345.databases.neo4j.io', path='', params='', query='', fragment=''), client_certificate = None`
seems to indicate that it parsed the uri scheme correctly. Which is what makes the error all the more confusing.

It seems like this is just a bug. 


```python
cls = <class 'neo4j._sync.driver.GraphDatabase'>, uri = 'neo4j+ssc://12345.databases.neo4j.io', auth = <neo4j._sync.auth_management.StaticAuthManager object at 0x10aad42f0>
config = {'auth': <neo4j._sync.auth_management.StaticAuthManager object at 0x10aad42f0>, 'trusted_certificates': <neo4j._conf.TrustCustomCAs object at 0x10aa13b60>}
driver_type = 'DRIVER_NEO4J', security_type = 'SECURITY_TYPE_SELF_SIGNED_CERTIFICATE'
parsed = ParseResult(scheme='neo4j+ssc', netloc='12345.databases.neo4j.io', path='', params='', query='', fragment=''), client_certificate = None

    @classmethod
    def driver(
        cls,
        uri: str,
        *,
        auth: _TAuth | AuthManager = None,
        **config,
    ) -> Driver:
        """
        Create a driver.
    
        :param uri: the connection URI for the driver,
            see :ref:`uri-ref` for available URIs.
        :param auth: the authentication details,
            see :ref:`auth-ref` for available authentication details.
        :param config: driver configuration key-word arguments,
            see :ref:`driver-configuration-ref` for available
            key-word arguments.
        """
        driver_type, security_type, parsed = parse_neo4j_uri(uri)
    
        if not isinstance(auth, AuthManager):
            auth = AuthManagers.static(auth)
        config["auth"] = auth
    
        client_certificate = config.get("client_certificate")
        if isinstance(client_certificate, ClientCertificate):
            # using internal class until public factory is GA:
            # AsyncClientCertificateProviders.static
            config["client_certificate"] = (
                _StaticClientCertificateProvider(client_certificate)
            )
    
        # TODO: 6.0 - remove "trust" config option
        if "trust" in config and config["trust"] not in {
            TRUST_ALL_CERTIFICATES,
            TRUST_SYSTEM_CA_SIGNED_CERTIFICATES,
        }:
            raise ConfigurationError(
                "The config setting `trust` values are {!r}".format(
                    [
                        TRUST_ALL_CERTIFICATES,
                        TRUST_SYSTEM_CA_SIGNED_CERTIFICATES,
                    ]
                )
            )
    
        if "trusted_certificates" in config and not isinstance(
            config["trusted_certificates"], TrustStore
        ):
            raise ConfigurationError(
                'The config setting "trusted_certificates" must be of '
                "type neo4j.TrustAll, neo4j.TrustCustomCAs, or"
                "neo4j.TrustSystemCAs but was {}".format(
                    type(config["trusted_certificates"])
                )
            )
    
        if security_type in {
            SECURITY_TYPE_SELF_SIGNED_CERTIFICATE,
            SECURITY_TYPE_SECURE,
        } and (
            "encrypted" in config
            or "trust" in config
            or "trusted_certificates" in config
            or "ssl_context" in config
        ):
            # TODO: 6.0 - remove "trust" from error message
>           raise ConfigurationError(
                'The config settings "encrypted", "trust", '
                '"trusted_certificates", and "ssl_context" can only be '
                "used with the URI schemes {!r}. Use the other URI "
                "schemes {!r} for setting encryption settings.".format(
                    [
                        URI_SCHEME_BOLT,
                        URI_SCHEME_NEO4J,
                    ],
                    [
                        URI_SCHEME_BOLT_SELF_SIGNED_CERTIFICATE,
                        URI_SCHEME_BOLT_SECURE,
                        URI_SCHEME_NEO4J_SELF_SIGNED_CERTIFICATE,
                        URI_SCHEME_NEO4J_SECURE,
                    ],
                )
            )
E           neo4j.exceptions.ConfigurationError: The config settings "encrypted", "trust", "trusted_certificates", and "ssl_context" can only be used with the URI schemes ['bolt', 'neo4j']. Use the other URI schemes ['bolt+ssc', 'bolt+s', 'neo4j+ssc', 'neo4j+s'] for setting encryption settings.

.venv/lib/python3.13/site-packages/neo4j/_sync/driver.py:251: ConfigurationError
```

--- a/docs/source/api.rst
+++ b/docs/source/api.rst
@@ -97,7 +97,9 @@ Each supported scheme maps to a particular :class:`neo4j.Driver` subclass that i
 
 
 .. note::
+
     See also :ref:`encryption-config-note-ref` to understand how the URI scheme relates to other encryption configuration options.
+
 
 .. note::
 
@@ -809,7 +811,7 @@ There are different *mutually exclusive* ways of configuring TLS/SSL encryption 
 
 * Use a URI scheme ending in ``+s``. This auto-configures the driver to use TLS and only trust system CAs.
 * Use a URI scheme ending in ``+ssc``. This auto-configures the driver to use TLS and trust any certificate.
-* Use any other URI scheme and
+* Use a URI scheme without suffix (i.e. ``neo4j://`` or ``bolt://``) and one of the following mutually exclusive options:
 
   * set :ref:`encrypted-ref` to ``True`` and optionally configure :ref:`trusted-certificates-ref` and/or
     :ref:`client-certificate-ref` to enable TLS with custom security settings.

--- a/docs/source/api.rst
+++ b/docs/source/api.rst
@@ -603,6 +603,9 @@ For example:
 ---------------
 Specify a custom SSL context to use for wrapping connections.
 
+The driver offers other, easier APIs for common encryption configurations (see :ref:`encryption-config-note-ref`).
+It's likely that your use-case doesn't actually require this options.
+
 This setting is only available for URI schemes ``bolt://`` and ``neo4j://`` (:ref:`uri-ref`).
 
 If given, ``encrypted``, ``trusted_certificates``, and ``client_certificate`` have no effect.

--- a/docs/source/api.rst
+++ b/docs/source/api.rst
@@ -95,6 +95,10 @@ Each supported scheme maps to a particular :class:`neo4j.Driver` subclass that i
 | neo4j+s                | :ref:`neo4j-driver-ref` with encryption (accepts only certificates signed by a certificate authority), full certificate checks.       |
 +------------------------+---------------------------------------------------------------------------------------------------------------------------------------+
 
+
+.. note::
+    See also :ref:`encryption-config-note-ref` to understand how the URI scheme relates to other encryption configuration options.
+
 .. note::
 
     See https://neo4j.com/docs/operations-manual/current/configuration/ports/ for Neo4j ports.
@@ -413,8 +417,11 @@ Additional configuration can be provided via the :class:`neo4j.Driver` construct
 + :ref:`user-agent-ref`
 + :ref:`driver-notifications-min-severity-ref`
 + :ref:`driver-notifications-disabled-categories-ref`
++ :ref:`driver-notifications-disabled-classifications-ref`
 + :ref:`driver-warn-notification-severity-ref`
 + :ref:`telemetry-disabled-ref`
+
+:ref:`encryption-config-note-ref`
 
 
 .. _connection-acquisition-timeout-ref:
@@ -792,6 +799,23 @@ The driver transmits the following information:
 :Default: :data:`False`
 
 .. versionadded:: 5.13
+
+
+.. _encryption-config-note-ref:
+
+Note on Encryption Configuration
+--------------------------------
+There are different *mutually exclusive* ways of configuring TLS/SSL encryption behavior of the driver:
+
+* Use a URI scheme ending in ``+s``. This auto-configures the driver to use TLS and only trust system CAs.
+* Use a URI scheme ending in ``+ssc``. This auto-configures the driver to use TLS and trust any certificate.
+* Use any other URI scheme and
+
+  * set :ref:`encrypted-ref` to ``True`` and optionally configure :ref:`trusted-certificates-ref` and/or
+    :ref:`client-certificate-ref` to enable TLS with custom security settings.
+  * or set :ref:`ssl-context-ref` to gain full control (and responsibility) over the TLS configuration.
+  * or set ``encrypted=False`` (default) to disable TLS.
+
 
 
 Driver Object Lifetime

--- a/docs/source/async_api.rst
+++ b/docs/source/async_api.rst
@@ -101,12 +101,16 @@ Each supported scheme maps to a particular :class:`neo4j.AsyncDriver` subclass t
 +------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
 | bolt+s                 | :ref:`async-bolt-driver-ref` with encryption (accepts only certificates signed by a certificate authority), full certificate checks.        |
 +------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
-| neo4j                  | :ref:`async-neo4j-driver-ref` with no encryption or with custom encryption configuration, see :ref:`async-driver-configuration-ref`.         |
+| neo4j                  | :ref:`async-neo4j-driver-ref` with no encryption or with custom encryption configuration, see :ref:`async-driver-configuration-ref`.        |
 +------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
 | neo4j+ssc              | :ref:`async-neo4j-driver-ref` with encryption (accepts self signed certificates).                                                           |
 +------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
 | neo4j+s                | :ref:`async-neo4j-driver-ref` with encryption (accepts only certificates signed by a certificate authority), full certificate checks.       |
 +------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
+
+
+.. note::
+    See also :ref:`encryption-config-note-ref` to understand how the URI scheme relates to other encryption configuration options.
 
 .. note::
 

--- a/docs/source/async_api.rst
+++ b/docs/source/async_api.rst
@@ -110,7 +110,9 @@ Each supported scheme maps to a particular :class:`neo4j.AsyncDriver` subclass t
 
 
 .. note::
+
     See also :ref:`encryption-config-note-ref` to understand how the URI scheme relates to other encryption configuration options.
+
 
 .. note::
 

--- a/src/neo4j/_conf.py
+++ b/src/neo4j/_conf.py
@@ -59,6 +59,9 @@ class TrustSystemCAs(TrustStore):
     certificate authority. This option is primarily intended for use with
     full certificates.
 
+    The driver loads the trusted system CAs using Python's
+    :meth:`ssl.SSLContext.load_default_certs`.
+
     For example::
 
         import neo4j

